doc/manual/tuto.md

```diff
@@ -0,0 +1,187 @@
+---
+slug: tutorial
+---
+
+# Managing users
+
+In this tutorial, you will learn how to use Nickel to manage a list of
+users, and then export it as a YAML file for a program consuming it.
+
+## Step 1: Install nickel
+
+The first step is to obtain nickel on your system. There are different
+methods to install Nickel, they are all covered on the [Getting
+started](https://nickel-lang.org/getting-started/#getting-started)
+page of Nickel project website.
+
+## Step 2: Think about the schema
+
+The YAML we want to produce has different fields with different types
+(lists, booleans, strings). Here is a sample of what we need:
+
+```yaml
+users:
+  - name: Aisha
+    is-admin: true
+    ssh-keys:
+      - AAAAApqCA8oKAB5S/47f...... aisha@work
+  - name: Violet
+    extra-groups:
+      - accounting
+```
+
+We can spot attributes such as `name`, `ssh-keys`, `is-admin` or
+`extra-groups`. To create a Nickel contract that correctly specifies
+the shape of this data, we need to think about the type needed for
+each attribute.
+
+For example, the attribute `name` is a string, which translates to `Str`
+in Nickel. Meanwhile, `ssh-keys` must allow multiple keys, so this is a
+list of strings, written as `Array Str` in Nickel. The field `is-admin`
+is a boolean, written as `Bool`. Finally, `extra-groups` is a list of
+group names, so we need `Array Str`, the same type used for `ssh-keys`.
+
+We can also mark fields as `optional` so you won't have to explicitly
+write them if they don't have any value. In the example,`extra-groups`
+and `ssh-keys` can be empty, so we will mark them as optional using the
+`optional` keyword.
+
+The field `is-admin` must always be present in the YAML file, but for
+most of our users it will be set to `false`. Fortunately, we can assign
+the default value `false` to this field , which means you only need
+to write `is-admin = true` in Nickel code when the user is actually
+an administrator.
+
+## Step 3: Write a contract
+
+Create a text file named `users-schema.ncl`. We will write our contract
+defining attributes and associated constraints such as a type, marking
+the attribute optional, and a default value if there is one.
+
+Please note that using a contract isn't mandatory per-se to use Nickel,
+but it will allow you to validate your input data (See Step 6 for an
+example).
+
+Please note that using a contract isn't mandatory per-se to use Nickel,
+but it will allow you to validate your input data (See Step 6 for an
+example).
+
+```nickel
+{
+  UserSchema =
+  {
+    name | Str,
+    ssh-keys | Array Str | optional,
```

Comment:
We don't have clear formatting rules yet (but soonish maybe, with topiary :slightly_smiling_face:  ) but I think we should abide by the rule of making a multiline block if there is more than one metadata; otherwise, I find it hard to read a bit overwhelming.

```suggestion
    ssh-keys
        | Array Str
        | optional,
```

Reply:
okay!